### PR TITLE
Lint hooks

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
Adds the excellent linting [package for hooks](https://www.npmjs.com/package/eslint-plugin-react-hooks).

## Why?
Because linting is great and so is this package

## What happened
We got linting errors straight away

## So
I fixed them. In particular, I refactored Onwards to move the data fetching into a separate `OnwardsData` component so that we are correctly using the hook at the top of the file and not wrapped it inside a conditional. Tut

## Link to supporting Trello card
https://trello.com/c/g9EpCWFd/1227-lint-our-usage-of-hooks

### CODE REVIEW

File: src/web/lib/useHover.tsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ export function useHover<T>(): [MutableRefObject<T>, boolean] {
                 node.removeEventListener('mouseout', handleMouseOut);
             };
         }
-    }, [ref.current]); // Recall only if ref changes
+    }, []); // Recall only if ref changes
```
Comment (liywjl) [inline src/web/lib/useHover.tsx]: why this change? Also remove comment

Comment (oliverlloyd) [reply src/web/lib/useHover.tsx]: Because:

```
React Hook useEffect has an unnecessary dependency: 'ref.current'. Either exclude it or remove the dependency array. Mutable values like 'ref.current' aren't valid dependencies because mutating them doesn't re-render the component.eslint(react-hooks/exhaustive-deps)
```

Thanks for the nudge on the comment 👍
